3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -17,27 +17,29 @@
 ------------------------------------------------------------------------------*/
 function requestData(url) {
   // TODO return a promise using `fetch()`
+  return fetch(url).then((response) => response.json());
```

Comment:
You should also check for HTTP errors, e.g.:

```js
if (!response.ok) {
  throw ...
}
```

Reply:
fetch(url).then((response) => {
    if (response.status >= 200 && response.status < 400) {
      console.log('we got it!')
        return response.json();
    }
    else {
        throw "HTTP ERROR";
    }
  });

Reply:
I am trying this but getting thus error:

Cannot read property 'img' of undefined.

Reply:
Rather than replying here it is better to make the changes to your code, adding comments if necessary, and push them. Then at least I can see what you have tried. Without seeing the code I can only guess.

Reply:
fetch(url).then((response) => {
    if (!response.ok) {
      throw "HTTP ERROR";
        
    }
    else {
        return response.json();
    }
  });
  
  I also tried this one but it returned the same error